libraries/chain/include/eosio/chain/backing_store/db_combined.hpp

```diff
@@ -209,76 +194,128 @@ class rocksdb_whole_db_table_collector {
                               std::size_t value_size) {
-      uint64_t primary_key;
-      EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, primary_key), bad_composite_key_exception,
-                 "DB intrinsic key-value store composite key is malformed");
-      backing_store::payer_payload pp(value, value_size);
-      receiver_.add_row(primary_index_view{primary_key, pp.payer, {std::string{pp.value, pp.value + pp.value_size}}});
+      if (!is_table_only()) {
+         uint64_t primary_key;
+         EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, primary_key), bad_composite_key_exception,
+                    "DB intrinsic key-value store composite key is malformed");
+         receiver_.add_row(primary_index_view::create(primary_key, value, value_size));
+      }
       ++primary_count_;
    }
 
    template<typename IndexType>
    void extract_secondary_index(b1::chain_kv::bytes::const_iterator remaining,
                                 b1::chain_kv::bytes::const_iterator key_end, const char *value,
                                 std::size_t value_size) {
-      IndexType secondary_key;
-      uint64_t primary_key;
-      EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, secondary_key), bad_composite_key_exception,
-                 "DB intrinsic key-value store composite key is malformed");
-      EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, primary_key), bad_composite_key_exception,
-                 "DB intrinsic key-value store composite key is malformed");
-      backing_store::payer_payload pp(value, value_size);
-      receiver_.add_row(secondary_index_view<IndexType>{primary_key, pp.payer, secondary_key});
+      if (!is_table_only()) {
+         IndexType secondary_key;
+         uint64_t primary_key;
+         EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, secondary_key), bad_composite_key_exception,
+                    "DB intrinsic key-value store composite key is malformed");
+         EOS_ASSERT(b1::chain_kv::extract_key(remaining, key_end, primary_key), bad_composite_key_exception,
+                    "DB intrinsic key-value store composite key is malformed");
+         backing_store::payer_payload pp(value, value_size);
+         receiver_.add_row(secondary_index_view<IndexType>{primary_key, pp.payer, secondary_key});
+      }
       ++secondary_count<IndexType>();
    }
 
    bool operator()(uint64_t contract, const char *key, std::size_t key_size,
                    const char *value, std::size_t value_size) {
-      if (!keep_processing_()) {
-         return false;
-      }
       b1::chain_kv::bytes composite_key(key, key + key_size);
       auto[scope, table, remaining, type] = backing_store::db_key_value_format::get_prefix_thru_key_type(
             composite_key);
 
+      const name code = name{contract};
+      if (table_context_ && is_reversed() && !is_same_table(code, scope, table)) {
+         table_context_->count = total_count();
```

Comment:
add comment:
// since the table is being traversed in reverse order and we have identified a key for a new table, need to add the key count for the completed table and report it

Reply:
Added.